assets/scss/components/_body-overlay.scss

```diff
@@ -0,0 +1,16 @@
+.c-body-overlay {
+    display: none;
+
+    .has-active-nav-pages & {
```

Comment:
Ideally `.is-active` applied on `.c-body-overlay` would be enough, but since you are aiming for nesting here, maybe use more general classname like `.overlay-shown` ?